api/v1alpha1/nodepool_types.go

```diff
@@ -139,6 +139,14 @@ type NodePoolSpec struct {
 	// +optional
 	NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
 
+	// Arch is an optional parameter specifying the preferred processor architecture for the NodePool
```

Comment:
Can we update the comment? I wouldn't define this as an optional parameter. As the PR is implemented the field is required, meaning that the backend wouldn't be able to proceed if it is not populated.
The reason you declare it here as optional is because [there's a limitation on client side validation integration with open API machinery](https://github.com/kubernetes/kubernetes/issues/108768#issuecomment-1253912215), it does not default before validating so if it's not set as +optional it would fail.

Reply:
Ok I'll update the comment to this:
```
	// Arch is the preferred processor architecture for the NodePool
	// NOTE: This is set as optional to prevent validation from failing due to a limitation on client side validation with open API machinery:
	//	https://github.com/kubernetes/kubernetes/issues/108768#issuecomment-1253912215
```